receive_stream.go

```diff
@@ -205,9 +207,13 @@ func (s *receiveStream) readImpl(p []byte) (bool, int, error) {
 		// when a RESET_STREAM was received, the flow controller was already
 		// informed about the final byteOffset for this stream
 		if !s.cancelledRemotely {
-			if queueMaxStreamData := s.flowController.AddBytesRead(protocol.ByteCount(m)); queueMaxStreamData {
+			hasStream, hasConn := s.flowController.AddBytesRead(protocol.ByteCount(m))
```

Comment:
nit: may as well assign these to your vars `hasXWindowUpdate`

Reply:
Depends on how the loop executes / returns. I think it should be fine, but I wanted to be very explicit about not setting value from `true` back to `false`.